### PR TITLE
Feature/iboss

### PR DESCRIPTION
**Pull request by lzammit**
# Description

It add the support on Iboss with the Single Sign On. Documentation will come.
# Impacts

None
# Delete branch after merge

YES 
# NEWS file entries
## New Features
- SSO integration with Iboss

### CODE REVIEW

File: lib/pf/firewallsso/Iboss.pm [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+            # Pass request to the user agent and get a response back
+            my $res = $ua->request($req);
+
+            # Check the outcome of the response
+            if ($res->is_success) {
+                $logger->info("Username $username with node $mac is unregistered and logout from the Iboss");
+            }
+            else {
+            Dumper('Logon: $req');
```
Comment (fdurand) [inline lib/pf/firewallsso/Iboss.pm]: remove it

File: lib/pf/firewallsso/Iboss.pm [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+use HTTP::Request::Common;
+
+#Export environement variables for LWP
+$ENV{'PERL_LWP_SSL_VERIFY_HOSTNAME'} = 0;
+
+=head1 METHODS
+
+=head2 action
+
+Perform a xml api request based on the registered status of the node and his role.
```
Comment (fdurand) [inline lib/pf/firewallsso/Iboss.pm]: it look like a GET not xml api request

File: lib/pf/firewallsso/Iboss.pm [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+=cut
+
+
+use strict;
+use warnings;
+
+use base ('pf::firewallsso');
+
+use POSIX;
+use Log::Log4perl;
```
Comment (fdurand) [inline lib/pf/firewallsso/Iboss.pm]: use pf::log;

File: lib/pf/firewallsso/Iboss.pm [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+Perform a xml api request based on the registered status of the node and his role.
+
+=cut
+
+my $ua = LWP::UserAgent->new;
+
+sub action {
+    my ($self,$firewall_conf,$method,$mac,$ip,$timeout) = @_;
+    my $logger = Log::Log4perl::get_logger(ref($self));
```
Comment (fdurand) [inline lib/pf/firewallsso/Iboss.pm]: my $logger = get_logger();

File: html/pfappserver/lib/pfappserver/Form/Config/Firewall_SSO/Iboss.pm [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Form::Config::Firewall_SSO';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::config;
+use pf::util;
+use File::Find qw(find);
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/Firewall_SSO/Iboss.pm]: use pf::log;

**Pull request merged**